### ISSUE TITLE
SSD training failes

### ISSUE THREAD

**Issue by bmalnar** [intel/caffe#135]
I cloned the repo today and tried to run SSD training as described on the Intel Caffe SSD wiki. I am using CentOS 7 on Knights Landing. I created the datasets and I am executing the command as instructed:

./build/tools/caffe train -solver models/intel_optimized_models/ssd/VGGNet/VOC0712/SSD_300x300/solver.prototxt -weights models/intel_optimized_models/ssd/VGGNet/VGG_ILSVRC_16_layers_fc_reduced.caffemodel

This results in the following error:

I1026 14:47:36.291285 145739 caffe.cpp:342] Starting Optimization
I1026 14:47:36.291457 145739 solver.cpp:559] Solving VGG_VOC0712_SSD_300x300_train
I1026 14:47:36.291539 145739 solver.cpp:560] Learning Rate Policy: multistep
I1026 14:47:46.261312 145739 solver.cpp:315] Iteration 0, loss = 23.0401
I1026 14:47:46.261502 145739 solver.cpp:336]     Train net output #0: mbox_loss = 23.0401 (* 1 = 23.0401 loss)
I1026 14:47:46.261606 145739 sgd_solver.cpp:190] Iteration 0, lr = 0.001
F1026 14:47:46.314705 145739 sgd_solver.cpp:518] Check failed: true == net_params[param_id]->get_prv_data_descriptor()->layout_compare( net_params[param_id]->get_prv_diff_descriptor()) (1 vs. 0)
*** Check failure stack trace: ***
    @     0x7f7d0fc1de6d  (unknown)
    @     0x7f7d0fc1fced  (unknown)
    @     0x7f7d0fc1da5c  (unknown)
    @     0x7f7d0fc2063e  (unknown)
    @     0x7f7d100fe50d  caffe::SGDSolver<>::Regularize()
    @     0x7f7d100fc36c  caffe::SGDSolver<>::ApplyUpdate()
    @     0x7f7d100fc287  caffe::SGDSolver<>::ApplyUpdate()
    @     0x7f7d102b326c  caffe::Solver<>::Step()
    @     0x7f7d102b357e  caffe::Solver<>::Solve()
    @     0x7f7d1908cf63  train()
    @     0x7f7d190866cc  main
    @     0x7f7d0c3b5b35  __libc_start_main
    @     0x7f7d1908738b  (unknown)
Aborted (core dumped)

**Comment by ftian1** [comment]
we are looking into this issue. Before we have feedback, you can use CAFFE engine at SSD train phase by adding "--engine=CAFFE" at your command line.

**Comment by chuanqi129** [comment]
I also encountered this problem when I am using resnet50 to do image classification on Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz. 

I1213 11:59:29.077236 154564 solver.cpp:559] Solving
I1213 11:59:29.077244 154564 solver.cpp:560] Learning Rate Policy: poly
I1213 12:00:07.635759 154564 solver.cpp:315] Iteration 0, loss = 6.90776
I1213 12:00:07.636054 154564 solver.cpp:336]     Train net output #0: loss = 6.90776 (* 1 = 6.90776 loss)
I1213 12:00:07.636085 154564 sgd_solver.cpp:190] Iteration 0, lr = 0.012
F1213 12:00:07.636147 154564 sgd_solver.cpp:516] Check failed: true == net_params[param_id]->get_prv_data_descriptor()->layout_compare( net_params[param_id]->get_prv_diff_descriptor()) (1 vs. 0)
*** Check failure stack trace: ***
    @     0x7fcd21b57a7d  google::LogMessage::Fail()
    @     0x7fcd21b5984c  google::LogMessage::SendToLog()
    @     0x7fcd21b5766c  google::LogMessage::Flush()
    @     0x7fcd21b5a15e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fcd2226c0bd  caffe::SGDSolver<>::Regularize()
    @     0x7fcd2226a07c  caffe::SGDSolver<>::ApplyUpdate()
    @     0x7fcd22269f97  caffe::SGDSolver<>::ApplyUpdate()
    @     0x7fcd2225ae0c  caffe::Solver<>::Step()
    @     0x7fcd2225b11e  caffe::Solver<>::Solve()
    @     0x7fcd2b02e01b  train()
    @     0x7fcd2b02761c  main
    @     0x7fcd1e6a2c05  __libc_start_main
    @     0x7fcd2b0283ad  (unknown)

**Comment by ftian1** [comment]
it's a bug in mkldnn_memory.cpp and will be fixed in next release. you can replace the layer engine to caffe at first if you know which layer raised this issue.

**Comment by hshen14** [comment]
Please wait for v1.1.0 release which solves the issue.
